packages/ses/src/module-load.js

```diff
@@ -507,18 +507,72 @@ const memoizedLoadWithErrorAnnotation = (
   return moduleLoading;
 };
 
+/**
+ * @template {readonly any[]} [Args=unknown[]]
+ * @callback SyncJob
+ * @param {...Args} args
+ * @returns {void}
+ */
+
+/**
+ * @template {readonly any[]} [Args=unknown[]]
+ * @callback AsyncJob
+ * @param {...Args} args
+ * @returns {Promise<void>}
+ */
+
+/**
+ * @template {readonly any[]} [Args=unknown[]]
+ * @callback EnqueueSyncJobFn
+ * @param {SyncJob<Args>} func
+ * @param {readonly [...Args]} args // the only way to make the types match
+ * @returns {void}
+ */
+
+/**
+ * @template {readonly any[]} [Args=unknown[]]
+ * @callback EnqueueAsyncJobFn
+ * @param {AsyncJob<Args>} func
+ * @param {readonly [...Args]} args // the only way to make the types match
```

Comment:
Note that despite using templates, if I mess up the arguments array ([here](https://github.com/endojs/endo/pull/2808/files#diff-f9990051733b98f01f6918095f0f2d4acaa3885a93f71dbafeeefc83ec2b8137R678)) to not match the type expected by `func` I still get no warning :(